src/verify-and-receive.ts

```diff
@@ -26,9 +27,15 @@ export async function verifyAndReceive(
     );
   }
 
-  return state.eventHandler.receive({
-    id: event.id,
-    name: event.name,
-    payload: JSON.parse(event.payload),
-  });
+  try {
+    return state.eventHandler.receive({
+      id: event.id,
+      name: event.name,
+      payload: JSON.parse(event.payload),
```

Comment:
The `try/catch` should be around just the `JSON.parse` because if `eventHandler.receive` throws an error that might not be valid JSON (I'm not sure if this can end up calling user provided functions, but even if it doesn't the cost of just catching `JSON.parse` is very cheap)